biorxiv/word_vector_experiment/07_biorxiv_visualize_embeddings_pretrained.py

```diff
@@ -0,0 +1,195 @@
+
+# +
+import itertools
+from pathlib import Path
+import re
+
+import pandas as pd
+import plotnine as p9
+# -
+
+journal_map_df = pd.read_csv(
+    "../exploratory_data_analysis/output/biorxiv_article_metadata.tsv", sep="\t"
+)
+journal_map_df.head()
+
+# Keep only current versions of articles
+biorxiv_umap_models_latest = {
+    "original": (
+        pd.read_csv("output/embedding_output/umap/biorxiv_umap_300.tsv", sep="\t")
+        .groupby("doi")
+        .agg(
+            {
+                "doi": "last",
+                "document": "last",
+                "umap1": "last",
+                "umap2": "last",
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+    "doc2vec": (
+        pd.read_csv(
+            "output/embedding_output/umap/biorxiv_umap_300_doc2vec.tsv", sep="\t"
+        )
+        .groupby("doi")
+        .agg(
+            {
+                "doi": "last",
+                "document": "last",
+                "umap1": "last",
+                "umap2": "last",
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+    "pretrained": (
+        pd.read_csv(
+            "output/embedding_output/umap/biorxiv_umap_300_pretrained.tsv", sep="\t"
+        )
+        .groupby("doi")
+        .agg(
+            {
+                "doi": "last",
+                "document": "last",
+                "umap1": "last",
+                "umap2": "last",
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+}
+
+# Keep only current versions of articles
+biorxiv_pca_models_latest = {
+    "original": (
+        pd.read_csv("output/embedding_output/pca/biorxiv_pca_300.tsv", sep="\t")
+        .groupby("doi")
+        .agg(
+            {
+                "doi": "last",
+                "document": "last",
+                "pca1": "last",
+                "pca2": "last",
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+    "doc2vec": (
+        pd.read_csv("output/embedding_output/pca/biorxiv_pca_300_doc2vec.tsv", sep="\t")
+        .groupby("doi")
+        .agg(
+            {
+                "doi": "last",
+                "document": "last",
+                "pca1": "last",
+                "pca2": "last",
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+    "pretrained": (
+        pd.read_csv(
+            "output/embedding_output/pca/biorxiv_pca_300_pretrained.tsv", sep="\t"
+        )
+        .groupby("doi")
+        .agg(
+            {
+                "doi": "last",
+                "document": "last",
+                "pca1": "last",
+                "pca2": "last",
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+}
+
+# # UMAP of the Documents
```

Comment:
Would you mind adding a brief description about what the difference is between these visualizations:
So you're visualizing articles from different categories using 1) document embeddings, 2) doc2vec embeddings or 3) word2vec embeddings. What is the difference between in how these methods embed the document? Looks like doc2vec has less separation but word2vec is similar to your original embedding. I'm not sure how much the dimension matters either.

Reply:
>So you're visualizing articles from different categories using 1) document embeddings, 2) doc2vec embeddings or 3) word2vec embeddings. What is the difference between in how these methods embed the document? Looks like doc2vec has less separation but word2vec is similar to your original embedding. I'm not sure how much the dimension matters either.

I'll update the notebook with a description, but to answer your question: 
1) This is my original approach where I'm taking the average of every word vector present within a document. 
2) Doc2vec generates a document vector and word vectors at the same time. It's an updated word2vec model where every sentence is tagged with a document id and each document has their own randomly initialized vector. As the neural network processes each sentence it updates the corresponding document vector. So once training is done, each document has a "representation" vector and every word has their own "representation" vector.  
3) This approach is the same as (1) with the exception that the model is initialized with an already trained word2vec network. This is called pretraining. Once each vector has been initialized proceed to train word2vec normally. Within the NLP field pretraining can help when your corpus is not sufficiently large enough to train.

Reply:
Ah I see. So 2 and 3 are updating an embedding based on that task of finding likelihood that the word "love" fills in the sentence:  "I ____ cats"? Thanks for the clarification

Reply:
>Ah I see. So 2 and 3 are updating an embedding based on that task of finding likelihood that the word "love" fills in the sentence: "I ____ cats"? Thanks for the clarification

All three do actually.  2 has an extension where a document vector gets updated along with word vectors, while 3 is exactly the same model as 1 just vectors are instantiated with another model rather than randomly initialized.